Source/UnitTests/Common/MutexTest.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <mutex>
+#include <thread>
+
+#include "Common/Mutex.h"
+
+template <typename MutexType>
+static void DoAtomicMutexTest()
+{
+  MutexType work_mtx;
+
+  bool worker_done = false;
+
+  static constexpr auto SLEEP_TIME = std::chrono::microseconds{1};
+
+  // lock() on main thread, unlock() on worker thread.
+  std::thread thread{[&, lk = std::unique_lock{work_mtx}] {
+    std::this_thread::sleep_for(SLEEP_TIME);
+    worker_done = true;
+  }};
+
+  // lock() waits for the thread to unlock().
+  if (std::lock_guard lk{work_mtx}; true)
+    EXPECT_TRUE(worker_done);
```

Comment:
This is a construction I'm not used to. Could it instead be expressed like this?

```suggestion
  {
    std::lock_guard lk{work_mtx};
    EXPECT_TRUE(worker_done);
  }
```